### PR TITLE
Register types related to K8s Ingress

### PR DESCRIPTION
**Pull request by michi-covalent**
please review commit by commit:

- 689581540b1de6d03d089bdfbd61e9c18d4f1aa1 Add types related to K8s Ingress.
- 7186ac70e4e0efeb7d989fe806564246e4456777 Generate k8s client and register ingress types.

### CODE REVIEW

File: pkg/k8s/slim/k8s/api/networking/v1/types.go [hunk 1]
```diff
@@ -191,3 +191,349 @@ type NetworkPolicyList struct {
...
+	// "Cluster".
+	// +optional
+	// +featureGate=IngressClassNamespacedParams
+	Namespace *string `json:"namespace,omitempty" protobuf:"bytes,5,opt,name=namespace"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IngressClassList is a collection of IngressClasses.
+type IngressClassList struct {
```
Comment (aanm) [inline pkg/k8s/slim/k8s/api/networking/v1/types.go:531]: This, similarly to `IngressClass`, `Ingress` and `IngressList`, needs to be registered in the file pkg/k8s/slim/k8s/api/networking/v1/register.go

**Review by aanm — CHANGES_REQUESTED** [review]
Just double checking, will we use all fields of all structures? If no, might make sense to remove certain fields.

Also, it would be nice to have the code that will use these structures in this same PR instead of different PRs.

File: pkg/k8s/slim/k8s/api/networking/v1/types.go (hunk 1 above)
Comment (michi-covalent) [reply pkg/k8s/slim/k8s/api/networking/v1/types.go:531]: yeah i did that in the next commit: https://github.com/cilium/cilium/blob/7186ac70e4e0efeb7d989fe806564246e4456777/pkg/k8s/slim/k8s/api/networking/v1/register.go#L35

would it be easier to review if it's a single commit?

**Comment by michi-covalent** [comment]
> Just double checking, will we use all fields of all structures? If no, might make sense to remove certain fields.

i'd say yes we will use all these fields in the future.

> Also, it would be nice to have the code that will use these structures in this same PR instead of different PRs.

that's yet to be written :joy_cat: i'm trying to keep each PR to a reasonable size, it's already +6k lines although it's mostly auto-generated code...

File: operator/watchers/envoy_config.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+			Services: []*v2alpha1.ServiceListener{
+				{
+					Name:      getServiceNameForIngress(ingress),
+					Namespace: ingress.Namespace,
+					Listener:  ingress.Name,
+				},
+			},
+			BackendServices: backendServices,
+			// TODO(michi) what is this
+			// Ingress:         true,
```
Comment (jrajahalme) [inline operator/watchers/envoy_config.go:47]: This is a flag for *proxy ingress to pod* vs. *proxy egress from pod*. This is not to be used with k8s Ingress :-). Also, it is yet to be seen if we will ever actually need CiliumEnvoyConfig for pod ingress. So far it is completely untested and the bpf datapath support for it is also missing.

**Comment by michi-covalent** [comment]
moved to https://github.com/cilium/cilium/pull/18127. closing

**Pull request closed without merging**